js/js_notwasm.go

```diff
@@ -191,14 +229,37 @@ func (v Value) InstanceOf(t Value) bool {
 	return instanceOf.Invoke(v, t).Bool()
 }
 
+func (v Value) Type() Type {
+	switch v.v {
+	case js.Undefined:
+		return TypeUndefined
+	case nil:
+		return TypeNull
+	}
+	switch typeOf.Invoke(v.v).String() {
```

Comment:
This works, but if you want better performance you probably want to avoid the string comparison. If you have `typeof(x) === "number"` in JavaScript (comparison with string constant), then the JS engines will recognize this and check directly for "x is of type number", skipping the intermediate string.

Reply:
Done. Thank you!

As I wanted to avoid importing `fmt` or other libs, I embedded the Type values in JS directly.